service/src/main/java/bio/terra/pipelines/app/controller/PipelinesApiController.java

```diff
@@ -170,11 +176,30 @@ public ResponseEntity<ApiGetJobsResponse> getPipelineJobs(
     EnumeratedJobs enumeratedJobs =
         jobService.enumerateJobs(userId, limit, pageToken, validatedPipelineName);
 
-    ApiGetJobsResponse result = JobApiUtils.mapEnumeratedJobsToApi(enumeratedJobs);
+    ApiGetJobsResponse result = jobApiUtils.mapEnumeratedJobsToApi(enumeratedJobs);
 
     return new ResponseEntity<>(result, HttpStatus.OK);
   }
 
+  @Override
+  public ResponseEntity<ApiCreateJobResponse> getPipelineJobResult(
+      @PathVariable("pipelineName") String pipelineName, @PathVariable("jobId") UUID jobId) {
+    final SamUser userRequest = getAuthenticatedInfo();
+    String userId = userRequest.getSubjectId();
+    PipelinesEnum validatedPipelineId = validatePipelineName(pipelineName);
+
+    JobApiUtils.AsyncJobResult<String> jobResult =
+        jobApiUtils.retrieveAsyncJobResult(jobId, userId, validatedPipelineId, String.class, null);
+
+    ApiCreateJobResponse response =
+        new ApiCreateJobResponse()
+            .jobReport(jobResult.getJobReport())
+            .errorReport(jobResult.getApiErrorReport())
+            .pipelineOutput(jobResult.getResult()); // this is null unless the job succeeded
```

Comment:
does our flight return anything as a `pipelineOutput` when it is successful?

Reply:
our flight does not currently set anything to the RESPONSE field in the working map, so nothing is currently returned here, no.